model/apmevent.go

```diff
@@ -62,17 +62,21 @@ type APMEvent struct {
 
 	// Timestamp holds the event timestamp.
 	//
-	// See https://www.elastic.co/guide/en/ecs/current/ecs-base.html
+	// See https://www.elastic.co/guide/en/ecs/current/ecs-base.html#field-timestamp
 	Timestamp time.Time
 
-	// Labels holds labels to apply to the event.
+	// Labels holds labels to apply to the event.  While it's `map[string]interface{}`,
```

Comment:
The model types should be closely resemble the document structure, so we can automatically generate the encoding. We should have two fields: `Labels` and `NumericLabels`.

Would it be reasonable to update this PR to split the labels when translating from incoming events to model types? i.e. in modeldecoder and processor/otel.

Reply:
I had implemented it like you mention at first but then changed it back to just reuse the `common.MapStr` since it required less changes to the codebase. You raise a valid point that models should be as close to the stored documents, so I think it makes sense to change it back to use two fields.